runtime-config.md

```diff
@@ -12,27 +12,28 @@ Presently there are `Prestart`, `Poststart` and `Poststop`.
 Hooks allow one to run code before/after various lifecycle events of the container.
 Hooks MUST be called in the listed order.
 The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+All hooks execute in the host environment (e.g. the same namespace, cgroups, etc. that apply to the host process).
 
 Hook paths are absolute and are executed from the host's filesystem.
 
 ### Pre-start
 
-The pre-start hooks are called after the container process is spawned, but before the user supplied command is executed.
+The pre-start hooks are called [after the container process is spawned, but before the user supplied command is executed](runtime.md#typical-lifecycle).
 They are called after the container namespaces are created on Linux, so they provide an opportunity to customize the container.
 In Linux, for e.g., the network namespace could be configured in this hook.
 
-If a hook returns a non-zero exit code, then an error including the exit code and the stderr is returned to the caller and the container is torn down.
+If a hook returns a non-zero exit code, [then an error including the exit code and the stderr is returned to the caller and the container is torn down](runtime.md#typical-lifecycle).
 
 ### Post-start
 
-The post-start hooks are called after the user process is started.
+The post-start hooks are called [after the user process is started](runtime.md#typical-lifecycle).
```

Comment:
Not really. If you bind-mount the ns files, you don't need a process to
hold the namespaces.

On Mon, Oct 12, 2015 at 4:02 PM, W. Trevor King <notifications@github.com>
wrote:

> In runtime-config.md
> <https://github.com/opencontainers/specs/pull/207#discussion_r41810455>:
>
> >
> >  ### Post-start
> >
> > -The post-start hooks are called after the user process is started.
> > +The post-start hooks are called [after the user process is started](runtime.md#typical-lifecycle).
>
> On Mon, Oct 12, 2015 at 03:48:06PM -0700, Vish Kannan wrote: One
> alternative to the runC behavior is that of unsharing namespaces (excepting
> pid), bind mounting the namespaces, and letting the hooks run before
> starting a new process in a new pid namespace (if requested by the user).
> In this scenario, the `container process` isn't required.
> In the “user requested a PID namespace” case, that sounds like “the
> container isn't fully setup” (since you don't have the requested PID
> namespace at pre-start time). In either case, there is *still* a container
> process with runtime-specified code at pre-start-hook time, because it's
> holding open the unshared namespaces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/specs/pull/207/files#r41810455>.
>